### DISCUSSION TITLE
Getting "Authenticating with `FIREBASE_TOKEN` is deprecated" error when authenticating with SERVICE ACCOUNT

### DISCUSSION THREAD

**Discussion by bernat-gomez** [codemagic-ci-cd/codemagic-docs#1841] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Add account service to env. variable
2. Configure firebase app distribution in ios build

      firebase:
        firebase_service_account: $GOOGLE_APPLICATION_CREDENTIALS
        ios:
          app_id: <some_ios_app_id>
          groups:
          - <some_group_in_firebase_distr> # Group in "Firebase Distr"

3. Launch build

Distribution fails for iOS, but works on Android using the same account service.

### Expected results

Successful upload to firebase distr.

### Actual results

The distribution phase fails with error:

`Publishing failed :|
Failed to publish /Users/builder/clone/build/ios/ipa/xyz.ipa to Firebase App Distribution: ⚠  Authenticating with `FIREBASE_TOKEN` is deprecated and will be removed in a future major version of `firebase-tools`. Instead, use a service account key with `GOOGLE_APPLICATION_CREDENTIALS`: https://cloud.google.com/docs/authentication/getting-started
i  uploading binary...

Error: failed to upload release. HTTP Error: 400, Request contains an invalid argument.
`

### Build id (optional)

64a2c134642c4702b200eb38

**Comment by dtrdic** [comment, marked as answer]
hey @bernat-gomez Firebase has marked authentication via token as deprecated and might disable it in future versions of firebase tool. But this is just a warning and should not cause the build to fail.

The main error here is `Request contains an invalid argument.`
Can you please double check the format of your appId that you entered inside your yaml file

```
publishing:
  firebase:
    ios:
      # Add your iOS app id retrieved from Firebase console in this format
      app_id: x:xxxxxxxxxxxx:ios:xxxxxxxxxxxxxxxxxxxxxx  
```

**Reply by bernat-gomez** [reply]
Thanks David! You're right, since iOS has 2 app ID's, it was using the wrong one.
